README.md

```diff
@@ -1314,6 +1315,98 @@ NOTE: Not all the rules are linked here, check [SwiftFormat](https://github.com/
 
   </details>
 
+* <a id='avoid-casting-protocols'></a>(<a href='#avoid-casting-protocols'>link</a>) **Avoid casting protocols to concrete types.**
+
+  <details>
+
+  #### Why?
+  Downcasting a `protocol` to a concrete type (eg `class`) can introduce potential errors when the type cannot be casted.  If you require methods on the concrete type not present in the `protocol`, amend the `protocol` to add them, or refactor the code to use the concrete type instead.
+
+  ```swift
+  // WRONG
+  protocol UserProvider {
+    // ...
+  }
+  final class UserManager: UserProvider {
+    //
+  }
+  
+  if let manager = provider as? UserManager {
+     // downcasted manager
+  }
```

Comment:
The idea here is that we would avoid the downcasting by using the object we need, or adding methods to the protocol, or making a new second protocol.

I would think you would want to refactor it to one of these more preferred alternatives:

```
protocol UserProvider {
     func doThing()
  }
  final class UserManager: UserProvider { }
  
  let provider: UserProvider
  
  provider.doThing()
  ```
  
  Or  
  ```
  protocol UserProvider {
  }
  final class UserManager: UserProvider { 
     func doThing()
  }
  
  let manager: UserManager
  
  manager.doThing()
  ```
  
  Or
  
   ```
  protocol UserProvider {
  }
  protocol UserManagerProtocol: UserProvider {
     func doThing()
  }
  final class UserManager: UserManagerProtocol { }
  
  let manager: UserManagerProtocol
  
  manager.doThing()
  ```
  
I intentionally left off the `// RIGHT` from the example because there's a so many ways to do this, but maybe I should add it?

Reply:
Of course there are times where downcasting is unavoidable (eg some third party library we don't own, or some Objc code is in the mix), but we would want to avoid it if we _can_ 🙂

Reply:
If that's the case I am all aboard